packages/nc-gui/components/project/spreadsheet/components/MoreActions.vue

```diff
@@ -220,6 +237,58 @@ export default {
         })
       );
     },
+    async exportXlsx() {
+      let offset = 0
+      let c = 1
+      try {
+        while (!isNaN(offset) && offset > -1) {
+          let res;
+          if (this.publicViewId) {
+            res = await this.$api.public.csvExport(this.publicViewId, ExportTypes.CSV_DATA, {
+              query: {
+                fields:
+                  this.queryParams &&
+                  this.queryParams.fieldsOrder &&
+                  this.queryParams.fieldsOrder.filter(c => this.queryParams.showFields[c]),
+                offset,
+                sortArrJson: JSON.stringify(
+                  this.reqPayload &&
+                    this.reqPayload.sorts &&
+                    this.reqPayload.sorts.map(({ fk_column_id, direction }) => ({
+                      direction,
+                      fk_column_id,
+                    }))
+                ),
+                filterArrJson: JSON.stringify(this.reqPayload && this.reqPayload.filters),
+              },
+              headers: {
+                'xc-password': this.reqPayload && this.reqPayload.password,
+              },
+            });
```

Comment:
You should handle shared view case as well. Nothing exported + no corresponding endpoint on backend.